nixos/modules/services/logging/journaldriver.nix

```diff
@@ -0,0 +1,112 @@
+
+{ config, lib, pkgs, ...}:
+
+with lib; let cfg = config.services.journaldriver;
+in {
+  options.services.journaldriver = {
+    enable = mkOption {
+      type        = types.bool;
+      default     = false;
+      description = ''
+        Whether to enable journaldriver to forward journald logs to
+        Stackdriver Logging.
+      '';
+    };
+
+    logLevel = mkOption {
+      type        = types.string;
+      default     = "info";
+      description = ''
+        Log level at which journaldriver logs its own output.
+      '';
+    };
+
+    logName = mkOption {
+      type        = with types; nullOr string;
+      default     = null;
+      description = ''
+        Configures the name of the target log in Stackdriver Logging.
+        This option can be set to, for example, the hostname of a
+        machine to improve the user experience in the logging
+        overview.
+      '';
+    };
+
+    googleCloudProject = mkOption {
+      type        = with types; nullOr string;
+      default     = null;
+      description = ''
+        Configures the name of the Google Cloud project to which to
+        forward journald logs.
+
+        This option is required on non-GCP machines, but should not be
+        set on GCP instances.
+      '';
+    };
+
+    logStream = mkOption {
+      type        = with types; nullOr string;
+      default     = null;
+      description = ''
+        Configures the name of the Stackdriver Logging log stream into
+        which to write journald entries.
+
+        This option is required on non-GCP machines, but should not be
+        set on GCP instances.
+      '';
+    };
+
+    applicationCredentials = mkOption {
+      type        = with types; nullOr string;
+      default     = null;
+      description = ''
+        Path to the service account private key (in JSON-format) used
+        to forward log entries to Stackdriver Logging on non-GCP
+        instances.
+
+        This option is required on non-GCP machines, but should not be
+        set on GCP instances.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    users.extraUsers.journaldriver = {
+      isSystemUser = true;
+      createHome   = true;
+      home         = "/var/journaldriver";
+
+      # for journal access
+      group = "systemd-journal";
+    };
+
+    systemd.services.journaldriver = {
+      description = "Stackdriver Logging journal forwarder";
+      script      = "${pkgs.journaldriver}/bin/journaldriver";
+      wantedBy    = [ "multi-user.target" ];
+
+      serviceConfig = {
+        Restart = "always";
+        User    = "journaldriver";
+      };
+
+      environment = {
+        "RUST_LOG"                       = cfg.logLevel;
+        "LOG_NAME"                       = cfg.logName;
+        "LOG_STREAM"                     = cfg.logStream;
+        "GOOGLE_CLOUD_PROJECT"           = cfg.googleCloudProject;
+        "GOOGLE_APPLICATION_CREDENTIALS" = cfg.applicationCredentials;
```

Comment:
I think it would be nicer to not use the `"` here, `RUST_LOG = cfg.logLevel` works just fine.

Reply:
:+1: Updated